itest/asset_meta_test.go

```diff
@@ -101,3 +101,31 @@ func testAssetMeta(t *harnessTest) {
 	_, err := t.tapd.CancelBatch(ctxb, &mintrpc.CancelBatchRequest{})
 	require.NoError(t.t, err)
 }
+
+// testMintAssetWithPrecision tests that we're able to mint an asset with a
+// specific precision.
+func testMintAssetWithPrecision(t *harnessTest) {
+	mintName := "test-asset-precision"
+	jsonData := []byte(`{"field1": "value1", "field2": "value2"}`)
+
+	assetMint := mintrpc.MintAssetRequest{
+		Asset: &mintrpc.MintAsset{
+			AssetType: taprpc.AssetType_NORMAL,
+			Name:      mintName,
+			AssetMeta: &taprpc.AssetMeta{
+				Data: jsonData,
+				Type: taprpc.AssetMetaType_META_TYPE_JSON,
+			},
+			Amount:    500,
+			Precision: 2,
+		},
+	}
+	rpcSimpleAssets := MintAssetsConfirmBatch(
+		t.t, t.lndHarness.Miner.Client, t.tapd,
+		[]*mintrpc.MintAssetRequest{&assetMint},
+	)
+
```

Comment:
Should assert that the precision field has indeed been encoded.

Reply:
this is done indirectly by checking the MetaHash of the asset genesis with the generated MetaHash of the metadata in the request. This is nested inside the helpers, could surface here too

Reply:
Added an explicit check too